### PR TITLE
[Sprint 11] Implement api setting #1

### PR DESCRIPTION
**Pull request by HyperionsThs**
[Sprint 11] Implement api setting
Link : https://dev.framgia.com/redmine/issues/130814
Effect : N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    protected NotificationOrEmailSetting(Parcel in) {
+        mWorkspace = in.readByte() != 0;
+        mGroup = in.readByte() != 0;
+        mUserSpecialType = in.readByte() != 0;
+    }
+
+    public static final Creator<NotificationOrEmailSetting> CREATOR =
+            new Creator<NotificationOrEmailSetting>() {
+                @Override
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java]: ERROR: 'method def modifier' have incorrect indentation level 16, expected level should be one of the following: 8, 12.

File: app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                public NotificationOrEmailSetting createFromParcel(Parcel in) {
+                    return new NotificationOrEmailSetting(in);
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java]: ERROR: 'method def' child have incorrect indentation level 20, expected level should be one of the following: 12, 16.

File: app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                }
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java]: ERROR: 'method def rcurly' have incorrect indentation level 16, expected level should be one of the following: 8, 12.

File: app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+                @Override
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java]: ERROR: 'method def modifier' have incorrect indentation level 16, expected level should be one of the following: 8, 12.

File: app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                public NotificationOrEmailSetting[] newArray(int size) {
+                    return new NotificationOrEmailSetting[size];
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java]: ERROR: 'method def' child have incorrect indentation level 20, expected level should be one of the following: 12, 16.

File: app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                }
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java]: ERROR: 'method def rcurly' have incorrect indentation level 16, expected level should be one of the following: 8, 12.

File: app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+            };
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/NotificationOrEmailSetting.java]: ERROR: 'object def rcurly' have incorrect indentation level 12, expected level should be one of the following: 4, 8.

File: app/src/main/java/com/framgia/wsm/screen/setting/SettingProfileViewModel.java [hunk 8]
```diff
@@ -59,10 +68,47 @@ public void onGetUserError(BaseException exception) {
...
+        mCurrentGroupPosition =
+                searchCurrentPositionGroup(mUser.getGroups(), mUser.getSetting().getGroupDefault());
+        setGroupNotifi(mUser.getSetting().getNotificationSetting().isGroup());
+        setWorkspaceNotifi(mUser.getSetting().getNotificationSetting().isWorkspace());
+        setUserSpecialTypeNotifi(mUser.getSetting().getNotificationSetting().isUserSpecialType());
+        setGroupEmail(mUser.getSetting().getEmailSetting().isGroup());
+        setWorkspaceEmail(mUser.getSetting().getEmailSetting().isWorkspace());
+        setUserSpecialTypeEmail(mUser.getSetting().getEmailSetting().isUserSpecialType());
+        setNotificationAll(isWorkspaceNotifi() && isUserSpecialTypeNotifi() && isGroupNotifi());
+        setEmailAll(isWorkspaceEmail() && isUserSpecialTypeEmail() && isGroupEmail());
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/setting/SettingProfileViewModel.java]: (!)

Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/setting/SettingProfileViewModel.java]: pls move all method to ``Setting`` model and using Setting model binding to xml

**Pull request merged**